scripts/setup/docker.sh

```diff
@@ -0,0 +1,220 @@
+
+
+    # If the channel line is not empty, assume 'rust-toolchain' is in TOML format.
+    # Otherwise, assume it is in plain text format (single line with version).
+    local version=""
+    if [ -n "${channel_line}" ]; then
+        # Try to extract the channel from TOML format.
+        # Get the value after '=' and remove spaces and quotes.
+        version=$(echo "${channel_line}" | cut -d'=' -f2 | tr -d '[:space:]' | tr -d '"')
+    else
+        # Fallback: assume plain text format (single line with version).
+        version=$(head -n 1 "${rust_toolchain_file}" | tr -d '[:space:]')
+    fi
+
+    rm -f "${rust_toolchain_file}"
+
+    # Check if the version string is empty.
+    if [ -z "${version}" ]; then
+        print_error "Failed to extract Rust version from 'rust-toolchain' file."
+        echo ""
+        return
+    fi
+
+    echo "${version}"
+}
+
+#==================================================================================================
+# Main Script
+#==================================================================================================
+#
+# DESCRIPTION
+#   Builds the Docker image for the Nanvix toolchain.
+#
+main() {
+    print_message "Building Docker image for Nanvix toolchain..."
+
+    # Sanity check that we are running inside a git repository.
+    if ! git rev-parse --is-inside-work-tree &> /dev/null; then
+        print_error "This script must be run inside a Git repository."
+        exit 1
+    fi
+
+    # Sanity check that branch name was set correctly.
+    if [ -z "${DEFAULT_BRANCH_NAME}" ]; then
+        print_error "Could not determine the default branch name."
+        exit 1
+    fi
+
+    # Sanity check that repository directory was set correctly.
+    if [ ! -d "${REPO_ROOT_DIR}" ]; then
+        print_error "Could not determine the repository root directory."
+        exit 1
+    fi
+
+    # Sanity check that docker is installed.
+    if ! command -v docker &> /dev/null; then
+        print_error "Docker is not installed."
+        exit 1
+    fi
+
+    # Sanity check that the Dockerfile exists.
+    if [ ! -f "${DOCKERFILE_PATH}" ]; then
+        print_error "Dockerfile not found at ${DOCKERFILE_PATH}"
+        exit 1
+    fi
+    print_message "Using Dockerfile at '${DOCKERFILE_PATH}'"
+
+
+    local rust_version
+    rust_version=$(get_rust_version)
+    print_message "Using Rust version: ${rust_version}"
+
+    # Sanity check if rust version was correctly set.
+    if [ -z "${rust_version}" ]; then
+        print_error "Could not determine Rust version from rust-toolchain file."
+        exit 1
+    fi
+
+    # Create logs directory if it doesn't exist.
+    mkdir -p "${REPO_LOGS_DIR}"
+
+    docker build --no-cache \
+        --progress=plain \
+        --build-arg RUST_VERSION="${rust_version}" \
+        --build-arg NANVIX_VERSION="${NANVIX_VERSION}" \
+        -t nanvix/toolchain "${REPO_SCRIPTS_DIR}/setup/" \
```

Comment:
Is it intentional that the tag contains no version? I would have expected `nanvix/toolchain:VERSION`.

Reply:
Good idea.